### PR TITLE
Fix and enable flaky test TestQueryManager.testQueryCountMetrics 

### PR DESCRIPTION
**Pull request by ClarenceThreepwood**
Fixes ##20447

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-tests/src/test/java/com/facebook/presto/tests/TestQueryManager.java [hunk 1]
```diff
@@ -237,28 +237,29 @@ public void testQueryOutputSizeExceeded()
...
+    @Test(timeOut = 120_000L)
     public void testQueryCountMetrics()
             throws Exception
     {
         DispatchManager dispatchManager = queryRunner.getCoordinator().getDispatchManager();
-        // Create a total of 10 queries to test concurrency limit and
+        // Create a total of 6 queries to test concurrency limit and
         // ensure that some queries are queued as concurrency limit is 3
-        createQueries(dispatchManager, 10);
+        createQueries(dispatchManager, 6);
```
Comment (ClarenceThreepwood) [inline presto-tests/src/test/java/com/facebook/presto/tests/TestQueryManager.java:247]: 2-3 out of 10 queries were consistently failing for me on my local mac complaining of insufficient resources

**Comment by ClarenceThreepwood** [comment]
I'm seeing good robust results after this fix. I'm inclined to think the retry mechanism in #20687 may not be necessary 
a) reduces the number of concurrent queries 
b) introduces a synchronization point for the queries in flight before checking state

**Comment by tdcmeehan** [comment]
@ClarenceThreepwood can you modify this test to use the black hole connector (see #22307 and #22298)?  The idea is to use that connector to force queries to hang.  It might require a change to the test, which will improve the sequencing.

File: presto-tests/src/test/java/com/facebook/presto/tests/TestQueryManager.java [hunk 2]
```diff
@@ -57,12 +58,20 @@
...
     public void setUp()
             throws Exception
     {
-        queryRunner = TpchQueryRunnerBuilder.builder().build();
+        queryRunner = getSimpleQueryRunner();
+        queryRunner.installPlugin(new BlackHolePlugin());
+        queryRunner.createCatalog("blackhole", "blackhole");
+        // Black hole connectors do not have external metadata, which means that table metadata is local to each coordinator.
+        // So for the purposes of the test setup, create a local table per connector.
+        for (int i = 0; i < queryRunner.getCoordinators().size(); i++) {
```
Comment (tdcmeehan) [inline presto-tests/src/test/java/com/facebook/presto/tests/TestQueryManager.java:72]: The referenced tests use multiple coordinators, however this one just uses a single coordinator, so we can remove the comment and the for loop and just use `queryRunner.getCoordinator()`.

**Review by tdcmeehan — APPROVED** [review]

**Pull request merged**